addon/lib/metrics.js

```diff
@@ -160,34 +137,33 @@ const Metrics = module.exports = {
     const dataParsed = variantMaker.parseTests(JSON.parse(data));
 
     store.experimentVariants[subject] = dataParsed;
-    Metrics.experimentFeaturesChanged(subject, dataParsed);
     Events.emit(EVENT_SEND_VARIANTS, {
       data: JSON.stringify(dataParsed),
       subject: self.id
     });
   },
 
   onExperimentPing: function(ev) {
+    const timestamp = Date.now();
     const { subject, data } = ev;
     const dataParsed = JSON.parse(data);
 
-    // TODO: Map add-on ID (subject) to other pingTypes as necessary
-    const pingType = 'testpilottest';
-
-    if (store.experimentVariants && subject in store.experimentVariants) {
-      dataParsed.variants = store.experimentVariants[subject];
-    }
-
-    const payload = {
-      version: 1,
-      test: subject,
-      payload: dataParsed
-    };
+    AddonManager.getAddonByID(subject, addon => {
```

Comment:
I would be up to switching Tab Center to an event-driven model if that makes people's lives easier.  (I started off batching because we thought it would be nicer on the Test Pilot side, but now that Test Pilot has to handle events, perhaps using them in Tab Center wouldn't be so bad.)